src/ui/public/Vis/AggConfig.js

```diff
@@ -259,9 +259,14 @@ define(function (require) {
     };
 
     AggConfig.prototype.makeLabel = function () {
-      if (!this.type) return '';
-      var pre = (_.get(this.vis, 'params.mode') === 'percentage') ? 'Percentage of ' : '';
-      return pre += this.type.makeLabel(this);
+      var label = (_.get(this.vis, 'params.mode') === 'percentage') ? 'Percentage of ' : '';
+      label += this.type.makeLabel(this);
+
+      if (this.params.customLabel && this.params.customLabel !== '') {
+        label = this.params.customLabel;
```

Comment:
If you return early by moving this new conditional to the top of the function, then you don't need to make any changes to the rest of the function.  Something like:
```js
AggConfig.prototype.makeLabel = function () {
  if (this.params.customLabel && this.params.customLabel !== '') {
    return this.params.customLabel;
  }
  // rest of the function as it was before
}
```